RELEASE.md

```diff
@@ -0,0 +1,26 @@
+# Release Process
+
+The custom-metrics-apiserver library is released following Kubernetes releases.
+The major and minor versions of custom-metrics-apiserver are in sync with
+upstream Kubernetes and the patch version is reserved for changes in
+custom-metrics-apiserver that we want to release without having to wait for the
+next version of Kubernetes. 
+
+The process is as follow and should always be in sync with Kubernetes:
+
+1. A new Kubernetes minor version is released
+1. An issue is proposing a new release with a changelog containing the changes
+   since the last minor release
+1. At least one [OWNER](OWNERS) must LGTM this release
```

Comment:
approver? (both, approvers and reviewers are listed in the OWNERS file)

Reply:
makes sense :+1: